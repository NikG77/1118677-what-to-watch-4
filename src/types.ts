export interface FilmType {
  backgroundColor: string,
  director: string,
  description: string,
  duration: number,
  id: number,
  isFavorite: boolean,
  genre: string,
  pictureBackground: string,
  poster: string,
  previewVideo: string,
  ratingScore: number,
  releaseDate: number,
  ratingCount: number,
  src: string,
  starring:Array<string>,
  title: string,
  videoLink: string,
}


export interface UserInfoInterface {
  id: number;
  email: string;
  name: string;
  avatarUrl: string;
}


export interface Comment {
    author: string;
    date: number;
    id: string;
    idUser: number;
    rating: number;
    text: string;
}



