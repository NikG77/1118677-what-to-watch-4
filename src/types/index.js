import PropTypes from "prop-types";

export const filmsType = PropTypes.arrayOf(PropTypes.shape({
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  ratingScore: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
  genre: PropTypes.string.isRequired,
  releaseDate: PropTypes.number.isRequired,
  pictureBackground: PropTypes.string.isRequired,
}));

export const mainFilmType = PropTypes.shape({
  genre: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.number.isRequired,
  poster: PropTypes.string.isRequired,
  pictureBackground: PropTypes.string.isRequired,
});

export const filmType = PropTypes.shape({
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  ratingScore: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
  genre: PropTypes.string.isRequired,
  releaseDate: PropTypes.number.isRequired,
  pictureBackground: PropTypes.string.isRequired,
});


