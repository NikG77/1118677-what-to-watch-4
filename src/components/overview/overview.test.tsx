import * as React from "react";
import * as renderer from "react-test-renderer";
import Overview from "./overview";
import {FilmType} from "../../types";

const film: FilmType = {
  title: `One Flew Over the Cuckoo's Nest`,
  src: ``,
  poster: ``,
  ratingScore: 8,
  ratingCount: 200,
  director: ``,
  starring: [``, ``],
  genre: `Drama`,
  releaseDate: 2020,
  pictureBackground: ``,
  previewVideo: ``,
  duration: 100,
  backgroundColor: ``,
  videoLink: ``,
  description: ` `,
  isFavorite: false,
  id: 1,
};

it(`Should Overview render correctly`, () => {
  const tree = renderer
    .create(<Overview
      film={film}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
