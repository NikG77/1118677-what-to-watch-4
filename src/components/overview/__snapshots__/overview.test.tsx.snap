// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Overview render correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        200
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
       
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        , 
         and other
      </strong>
    </p>
  </div>,
]
`;
